### PR TITLE
Cp 3385/feature/dynamic button component

### PR DESCRIPTION
**Pull request by alexholmes99**
Created dynamic button component for form

### CODE REVIEW

File: src/Button.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import styled from "styled-components";
+
+function Button(props) {
+  return (
```
Comment (damonspencerroberts) [inline src/Button.js:5]: Nice job, might want to add some more dynamic behaviour.

You could add props for color, background, border, etc and then pass it to the styled component.

Example:
`props.buttonOptions = { background: 'red', color: 'blue', border: '1px solid grey' }`

```
<StyledButton type={props.type} value={props.value} buttonOptions={props.buttonOptions} >
      {props.content}
</StyledButton>

```

```
const StyledButton = styled.button`
  border: ${(props) => props.buttonOptions.border};
  opacity: 0.8;
  color: ${(props) => props.buttonOptions.color};
  font-size: 24px;
  width: 100%;
  font-family: "Sue Ellen Francisco", cursive;
  background: ${(props) => props.buttonOptions.background};
`;
```

You can also set default props to default to certain things if nothing is inputed. 

https://www.geeksforgeeks.org/reactjs-defaultprops/#:~:text=The%20defaultProps%20is%20a%20React,default%20props%20for%20the%20class.

File: src/Form.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        ...previousState,
+        [key]: value === "",
+      };
+    });
+  };
+
+  return (
+    <OuterForm>
+      <FormContent>
+        {emptyData.name && <h2>Name required</h2>}
```
Comment (damonspencerroberts) [inline src/Form.js:41]: Maybe add a styled component for the error message that's red...

File: src/Form.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        <TextArea
+          name="note"
+          placeholder="Enter Note Here"
+          value={formData.note}
+          onChange={(e) => handleChange(e, "note")}
+          onBlur={(e) => handleBlur(e.target.value, "note")}
+        />
+      </FormContent>
+      <FormContent>
+        <Button content="Submit" type="Submit" value="Submit" />
```
Comment (damonspencerroberts) [inline src/Form.js:72]: Why are you adding content prop?

**Review by damonspencerroberts — CHANGES_REQUESTED** [review]
Couple of things, overall starting to look very nice :)

File: src/Form.js (hunk 3 above)
Comment (alexholmes99) [reply src/Form.js:72]: So the label for the button can be changed so it doesn't always read "Submit"

**Pull request closed without merging**